### PR TITLE
Js to ts (first pass)

### PR DESCRIPTION
**Pull request by royendo**

### CODE REVIEW

File: src/pages/api/view-iframe.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+
+// Get the secret Rill service token from an environment variable.
+const rillServiceToken = process.env.RILL_SERVICE_TOKEN;
+
+// Configure the dashboard to request an iframe URL for.
+// Note that the organization must be the same as the one the service token is associated with.
+const rillOrg = "demo";
+const rillProject = "rill-openrtb-prog-ads";
+const rillDashboard = "auction_data_model_metrics_explore";
```
Comment (ericpgreen2) [inline src/pages/api/view-iframe.ts:10]: Note: we should rename the examples to use cleaner names, e.g. `openrtb` and `auction`

File: src/pages/api/nav-iframe.ts [hunk 2]
```diff
@@ -33,6 +35,6 @@ export default async function handler(req, res) {
             throw new Error(data.message);
         }
     } catch (error) {
-        res.status(500).json({ error: error.message });
+        res.status(500).json({ error: (error as Error).message });
```
Comment (ericpgreen2) [inline src/pages/api/nav-iframe.ts:38]: It's generally not recommended to cast variables, if you can avoid it. I see there are a lot of files with `error as Error`.

Here's another approach:
```
function getErrorMessage(error: unknown): string {
    if (error instanceof Error) return error.message;
    return String(error);
}

// Then in your catch block:
try {
    // ... existing code ...
} catch (error) {
    res.status(500).json({ error: getErrorMessage(error) });
}
```

**Review by ericpgreen2 — APPROVED** [review]
Nice!

File: src/pages/api/view-iframe.ts (hunk 1 above)
Comment (royendo) [reply src/pages/api/view-iframe.ts:10]: will make a new PR on this on rill-examples and follow up here afterwards

File: src/pages/api/nav-iframe.ts (hunk 2 above)
Comment (royendo) [reply src/pages/api/nav-iframe.ts]: created a utils and referenced in the apis!

**Comment by royendo** [comment]
will add the name changes to this PR, too

File: src/pages/api/view-iframe.ts (hunk 1 above)
Comment (royendo) [reply src/pages/api/view-iframe.ts]: i change the file names, do we want to change the project names too?

**Pull request merged**